src/runtime/cuda/cuda_device_api.cc

```diff
@@ -73,9 +74,9 @@ class CUDADeviceAPI final : public DeviceAPI {
         return;
       }
       case kDeviceName: {
-        cudaDeviceProp props;
-        CUDA_CALL(cudaGetDeviceProperties(&props, ctx.device_id));
-        *rv = std::string(props.name);
+        char name[256];
```

Comment:
consider directly create a std::string with the corresponding length and resize back to the size

Reply:
Get updated.